Add grammar validation for PPL

ykmr1224 opened the pull request:
### Description
- Add grammar validation for PPL
- Renamed the original class to differentiate SQL and PPL version

### Related Issues
n/a

### Check List
- [x] New functionality includes testing.
- [ ] New functionality has been documented.
 - [ ] New functionality has javadoc added.
 - [ ] New functionality has a user manual doc added.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
- [x] Commits are signed per the DCO using `--signoff`.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/sql/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

dai-chen reviewed (commented):
Are we merging these changes from other branch? I recall we've added PPL validator previously or I missed some context?

ykmr1224 commented:
> Are we merging these changes from other branch? I recall we've added PPL validator previously or I missed some context?

We added validator only for SQL, not PPL yet.

dai-chen commented:
Triggering CI again. Not sure if the CI failure is transient.
